### PR TITLE
S✔ ◾ Updating @octokit/plugin-rest-endpoint-methods to latest

### PR DESCRIPTION
**Pull request by muiriswoulfe**
### Motivation

There was a dependency on `@octokit/plugin-rest-endpoint-methods` version 6.2.0. This introduced risk as the dependency could develop security vulnerabilities over time, and it would not be possible to migrate to the latest version to resolve these.

### Technical

This dependency was updated to the latest version by changing the API calls to remove the now unsupported dependency on the GitHub diff position and to replace it with a diff line number. This involves reading the diff and parsing it to find the appropriate line number. New logic and dependencies have been introduced for this.

This change also includes:
- Some refactoring to support the above changes.
- Updating of all dependencies to the latest versions.
- Refactoring of integration tests into unit tests, as the integration tests stopped working due to a changed dependency that does not support test environments.
- Refactoring of the product code to enable the integration tests to be run as unit tests, due to the changes outlined above.
- Removal of superfluous test logic following the test refactoring.
- Onboarding of Babel to rewrite some dependencies to an older version of Node.js (v10), in order to continue supporting Azure DevOps. This rewriting is not performed for GitHub, where the version of Node.js used is much more recent.

### Test Types

- [X] Unit tests
- [X] Manual tests

### Unit Test Coverage

100%

### Other Test Details

The integration tests stopped working due to a changed dependency that can no longer reliably run within a test environment. These integration tests were converted to unit tests to avoid test logic being permanently lost.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -11,28 +11,33 @@
       "dependencies": {
```
Comment (muiriswoulfe) [inline package-lock.json:11]: ❗ **This file doesn't require review.**

File: src/task/src/git/octokitGitDiffParser.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      diffParsed.files.forEach((file: AnyFileChange): void => {
+        if (file.type === 'AddedFile' || file.type === 'ChangedFile') {
+          const fileCasted: AddedFile | ChangedFile = file as AddedFile | ChangedFile
+          result.set(fileCasted.path, fileCasted.chunks[0]!.toFileRange.start)
+        } else if (file.type === 'RenamedFile') {
+          const fileCasted: RenamedFile = file as RenamedFile
+          result.set(fileCasted.pathAfter, fileCasted.chunks[0]?.toFileRange.start!)
+        }
+      })
+    })
```
Comment (K-Cully) [inline src/task/src/git/octokitGitDiffParser.ts:76]: This code is difficult to read.
Suggest either adding comments or extracting some of the logic to well named functions that self-document,

File: src/task/src/pullRequestMetrics.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  /**
+   * Runs the overall PR Metrics task.
+   * @param folder The root folder containing index.ts.
+   * @returns A promise for awaiting the completion of the method call.
+   */
+  public async run (folder: string): Promise<void> {
+    try {
+      this._runnerInvoker.locInitialize(folder)
+
+      // TODO: Remove System.AccessToken support after a transition period has elapsed.
```
Comment (K-Cully) [inline src/task/src/pullRequestMetrics.ts:41]: Is there an issue logged for this?
Ideally this comment would reference the issue number.

Comment (muiriswoulfe) [reply src/task/src/pullRequestMetrics.ts:41]: Makes sense. I've created #304 and updated the comment here and in another file to reference the issue number.

File: src/task/src/git/octokitGitDiffParser.ts (hunk 2 above)
Comment (muiriswoulfe) [reply src/task/src/git/octokitGitDiffParser.ts:76]: Looking at this again, I agree with you. I've now refactored and commented as suggested and it should hopefully be easier to read now.

**Review by K-Cully — APPROVED** [review]

**Review by dimeehan — APPROVED** [review]

**Pull request merged**